### DISCUSSION TITLE
Entity Framework events in Cosmos DB

### DISCUSSION THREAD

**Discussion by pablotola** [thepirat000/Audit.NET#437] [Q&A]
Trying to query events in cosmos db filtering by the entity framework properties, but I get an exception in my query.
Basically it is saying that the GetEntityFrameworkEvent is not supported, below is my query, please advice on how to get the events querying the inner ef entry properties.

```
List<EventEntry> events = cosmosProvider.QueryEvents()
                .SelectMany(ev => ev.GetEntityFrameworkEvent().Entries)
                .Where(e => e.Name == entityName && e.PrimaryKey[primaryKey].ToString() == id)
                .Skip(page * pageSize)
                .Take(pageSize)
                .ToList();
```

**Comment by thepirat000** [comment]
I guess that's because it cannot be translated to a SQL query. Could you include the stack trace and the exception?

**Comment by thepirat000** [comment]
You could try with something like this:

```c#
List<EventEntry> events = cosmosProvider.QueryEvents<AuditEventEntityFramework>()
    .SelectMany(ev => ev.EntityFrameworkEvent.Entries)
    .Where(e => e.Name == entityName && e.PrimaryKey[primaryKey].ToString() == id)
    .Skip(page * pageSize)
    .Take(pageSize)
    .ToList();
```

But I'm not sure if the `SelectMany` will work
